### ISSUE TITLE
Completions: Extension host crashes when connected to a remote service

### ISSUE THREAD

**Issue by philipp-spiess** [sourcegraph/sourcegraph#53086]
Repro:

- Go to gitpod.io and start a new pod. Select "dekstop vscode"
- Install Cody into the remote pod
- (optional) Set up Cody and completions
- Write some code (to trigger a completion) and see that the process crashes

Logs:

We don't have logs from the _extension host_ processes yet. From the UI process, we can only observe this:

```
  ERR socket hang up: Error: socket hang up
	at connResetException (node:internal/errors:704:14)
	at TLSSocket.socketCloseListener (node:_http_client:454:25)
	at TLSSocket.emit (node:events:525:35)
	at node:net:757:14
	at TCP.done (node:_tls_wrap:584:7)
```

c.f

- https://github.com/sourcegraph/sourcegraph/pull/50047#issuecomment-1580979005
- https://sourcegraph.slack.com/archives/C052G9Y5Y8H/p1685730570636759

**Comment by jeanp413** [comment]
@philipp-spiess you can get the logs from the server by connecting using ssh from the terminal and then open the file `~/.vscode-server/.<HASH>.log`

**Comment by philipp-spiess** [comment]
@jeanp413 Amazing, thanks so much for the info. Will do more digging then. :)

**Comment by akosyakov** [comment]
You also should see all system log channels in Output view for affected window.

**Comment by philipp-spiess** [comment]
@akosyakov Oh I wasn't aware that you can access these basic logs through the window as well. It makes much sense in hindsight though 😐 

--- 

An update to the issue I was looking at:

- I can still repro the `socket hang up` error, nothing changed there.
- However, completions are still working just fine
- And any of the log sources I checked for extension host processes are quiet. I’m not sure if we really have a crash in the extension host here anymore.

I’m no longer sure if “socket hang up” is not just a red herring at this point. I can not reproduce the issue that we got reported unfortunlaty, @abeatrix. We'll have to work more with the user to get better logs. We know where to get them now, though 👀

**Comment by philipp-spiess** [comment]
This is fixed now thanks to #53956
